src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts

```diff
@@ -35,7 +34,16 @@ export class EthereumCleverAbcTokenFetcher extends AppTokenTemplatePositionFetch
     ];
   }
 
-  async getPricePerShare() {
-    return [(await this.getContract(abcCVX).ratio()).div(10 ** 6).toNumber() / 2];
+  async getPricePerShare({ } : GetPricePerShareParams<AbcCvx> )  {
+    return [1]; 
+  }
+
+  async getPrice({ }: GetPriceParams<AbcCvx>): Promise<number> {
+    return await axios.get('https://api.aladdin.club/api/aladdin/initInfo').then(v => v.data.data.abcCVXPrice)
```

Comment:
You shouldn't have to fetch prices from an external sources

Reply:
Okay, but how should I get it then? Since there is no dex pool or something

Reply:
O I see, the price will be calculated based on pricePerShare it it isn't overridden.

Reply:
All you need to do is fetch the underlying reserves, apply the formula below and bingo !

```
  async getUnderlyingTokenDefinitions(/*{ contract }: GetUnderlyingTokensParams<AbcCvx>*/) {
    return [
      { address: '0xf9078fb962a7d13f55d40d49c8aa6472abd1a5a6', network: this.network },
      { address: '0xf05e58fcea29ab4da01a495140b349f8410ba904', network: this.network },
    ];
  }

  async getPricePerShare({ appToken, contract }: GetPricePerShareParams<AbcCvx>) {
    const reserves = await Promise.all([contract.totalCurveLpToken(), contract.totalDebtToken()]);

    const pricePerShareRaw = reserves.map(r => Number(r) / appToken.supply);
    const pricePerShare = pricePerShareRaw.map(x => x / 10 ** appToken.decimals);

    return pricePerShare;
  }
```